### ISSUE TITLE
No comprehensive list of plugins with explanations of what they do

### ISSUE THREAD

**Issue by helpdata** [notepad-plus-plus/nppPluginList#158]
Just spent the last hour trying to find a full list of all NPP plugins with explanations of what each one does. There should be a readme.txt file somewhere with this information.

**Comment by chcg** [comment]
@helpdata This repo contains the input for the pluginAdmin window visible in N++ in json format containing a description for each plugin. The input therefore is:
https://github.com/notepad-plus-plus/nppPluginList/blob/master/src/pl.x64.json
and
https://github.com/notepad-plus-plus/nppPluginList/blob/master/src/pl.x86.json
where the raw data is available in text form.
Do you have other needs?

**Comment by akoppa** [comment]
If you mean a plugin list in plain English like is normal for any software project available for a general public: the answer is NO.

**Comment by helpdata** [comment]
I agree with akoppa. Digging into a json file is not an adequate answer.

**Comment by chcg** [comment]
So your expectation is something like https://marketplace.visualstudio.com/vscode or https://github.com/emacs-tw/awesome-emacs#awesome-emacs?

There are currently no plans for that, but I think it should be possible to transform the json input to a markdown file with the plugin name + description + homepage to have at least the basic informations.

**Comment by mattesh** [comment]
What about a generator to build it from those files?
Any language preference?

**Comment by chcg** [comment]
@mattesh Currently python is used to parse the json input against a schema and check that the downloads are usable. So python probably would be a good choice to create some markdown file from the x32 and x64 input files.

**Comment by mattesh** [comment]
...
Python -> md sounds good, but.
Please, don't expect smthng tomorrow ;-D

**Comment by mattesh** [comment]
What about this:
get_pl.py takes one parameter x**.json and prints the ready made table w/o header and footer.
Description "\r\n" gets replaced by &lt;br> and "|" by & vert;

Two tables of this format would be printed:
1. Plugin Name = ij.display-name
2. Author = ij.author
3. Homepage = ij.homepage
4. Version and link = "[" + ij.version + " - " + json_file.arch + " bit](" + ij.repository +")"
5. Description = " &lt;details> &lt;summary> " + first_two_lines(ij.description) + " &lt;/summary> " rest_of_text(ij.description) +"&lt;/details>"

## List of Plugins 32 bit

`|Plugin name | Author | Homepage | Version and link | Description |`
`|---|---|---|---|---|---|`
`| 3P - Progress Programmers Pal | Julien Caillon | [3P - 32 bit](https://jcaillon.github.io/3P/) | [1.8.6](https://github.com/jcaillon/3P/releases/download/v1.8.6/3P.zip) | <details> <summary> [Requires .NET framework v4.6.1]<br>Designed to help writing OpenEdge ABL (formerly known as Progress 4GL) code. It provides :</summary> <br>- a powerful auto-completion<br>- tool-tips on every words<br>- a code explorer to quickly navigate through your code<br>- a file explorer to easily access all your sources<br>- the ability to run/compile and even PROLINT your source file with an in-line visualization of errors<br>- more than 50 options to better suit your needs<br>- and so much more!<br>Visit https://jcaillon.github.io/3P/ for more details on the plugin </details> |`

**Comment by mattesh** [comment]
Hi,
what do you think about this?
https://github.com/notepad-plus-plus/nppPluginList/compare/6f40c133...mattesh:e709d1935a
[plugin_list_x64.zip](https://github.com/notepad-plus-plus/nppPluginList/files/4201722/plugin_list_x64.zip)

Only point to clarify would be how to copy it into stable place when commit was taken into the master.
Regards, Mattes

**Comment by chcg** [comment]
## Plugin List - 64 bit
version 1.1.7
|Plugin name | Author | Homepage | Version and link | Description |
|---|---|---|---|---|---|
| 3P - Progress Programmers Pal | Julien Caillon | https://jcaillon.github.io/3P/ | [1.8.6 - 64 bit](https://github.com/jcaillon/3P/releases/download/v1.8.6/3P_x64.zip) |  <details> <summary> [Requires .NET framework v4.6.1]<br>Designed to help writing OpenEdge ABL (formerly known as Progress 4GL) code. It provides : </summary> <br>- a powerful auto-completion<br>- tool-tips on every words<br>- a code explorer to quickly navigate through your code<br>- a file explorer to easily access all your sources<br>- the ability to run/compile and even PROLINT your source file with an in-line visualization of errors<br>- more than 50 options to better suit your needs<br>- and so much more!<br>Visit https://jcaillon.github.io/3P/ for more details on the plugin </details> |

**Comment by chcg** [comment]
@mattesh Lokes usable to me. In your example there is an
`|---|`
to much. Only drawback seen here is the collapsed description data which makes it difficult to search.

**Comment by mattesh** [comment]
@chcg 
the solution would be a small script snippet. But we seem to be not allowed to add this directly on [github](https://stackoverflow.com/questions/21340803/embed-javascript-in-github-readme-md).
We may be able embed some html script on the server: [from here](https://stackoverflow.com/questions/43008609/expanding-all-details-tags). However, my limited tests did not show a solution to combine script with markdown code.

Do you have already an idea how we would make this lists visible in some place?
e.g. in [plugin docs](https://npp-user-manual.org/docs/plugins/)?
Regards, Mattes

**Comment by chcg** [comment]
@mattesh My first though was to add the md files to the artefacts of a release, but then they are not immediately visible from within the browser.

**Comment by mattesh** [comment]
isn't it at the time when a new release is created, when the list need to be generated and pushed into https://github.com/notepad-plus-plus/npp-usermanual ?
See more in the other [thread.](https://github.com/notepad-plus-plus/userDefinedLanguages/issues/10#issuecomment-591173492)

**Comment by chcg** [comment]
First test versions from appveyor CI:
https://github.com/notepad-plus-plus/nppPluginList/blob/master/doc/plugin_list_x86.md
and
https://github.com/notepad-plus-plus/nppPluginList/blob/master/doc/plugin_list_x64.md
from current master. 
Still needs some automatic checkin which might be easier done by github action.

**Comment by chcg** [comment]
Updated manually with each tagging.
